test/distributed/org/apache/cassandra/distributed/test/FqlReplayDDLExclusionTest.java

```diff
@@ -0,0 +1,104 @@
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import com.datastax.driver.core.Session;
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.api.IInvokableInstance;
+import org.apache.cassandra.tools.ToolRunner;
+
+import static org.apache.cassandra.distributed.api.Feature.GOSSIP;
+import static org.apache.cassandra.distributed.api.Feature.NATIVE_PROTOCOL;
+import static org.apache.cassandra.distributed.api.Feature.NETWORK;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class FqlReplayDDLExclusionTest extends TestBaseImpl
+{
+    @Rule
+    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test
+    public void test() throws Throwable
+    {
+        try (final Cluster cluster = init(builder().withNodes(1)
+                                                   .withConfig(updater -> updater.with(NETWORK, GOSSIP, NATIVE_PROTOCOL))
+                                                   .start()))
+        {
+            final IInvokableInstance node = cluster.get(1);
+
+            // using driver path is important because dtest API and query execution does not invoke code
+            // in Cassandra where events are propagated to logger
+            try (com.datastax.driver.core.Cluster c = com.datastax.driver.core.Cluster.builder().addContactPoint("127.0.0.1").build();
+                 Session s = c.connect())
+            {
+                s.execute("CREATE KEYSPACE fql_ks WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};");
+
+                node.nodetool("enablefullquerylog", "--path", temporaryFolder.getRoot().getAbsolutePath());
+
+                s.execute("CREATE TABLE fql_ks.fql_table (id int primary key);");
+                s.execute("INSERT INTO fql_ks.fql_table (id) VALUES (1)");
+
+                node.nodetool("disablefullquerylog");
+            }
+
+            // here we are dropping and we expect that ddl replay will reconstruct it
+
+            node.executeInternal("DROP TABLE fql_ks.fql_table;");
+
+            final ToolRunner.Runners runners = new ToolRunner.Runners();
+
+            // without --replay-ddl-statements, the replay will fail on insert because underlying table is not there
+            final ToolRunner negativeRunner = runners.invokeClassAsTool("org.apache.cassandra.fqltool.FullQueryLogTool",
+                                                                        "replay",
+                                                                        "--keyspace", "fql_ks",
+                                                                        "--target", "127.0.0.1",
+                                                                        "--", temporaryFolder.getRoot().getAbsolutePath());
+
+            assertEquals(0, negativeRunner.getExitCode());
```

Comment:
if all queries fail we return 0?  Feels unexpected...

Reply:
yes, it does.